seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/source/KeyedRecordReader.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.seatunnel.common.utils.JsonUtils;
+import org.apache.seatunnel.connectors.seatunnel.redis.client.RedisClient;
+import org.apache.seatunnel.connectors.seatunnel.redis.config.RedisParameters;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+@Slf4j
+public class KeyedRecordReader extends RedisRecordReader {
+
+    public KeyedRecordReader(
+            RedisParameters redisParameters,
+            DeserializationSchema<SeaTunnelRow> deserializationSchema,
+            RedisClient redisClient) {
+        super(redisParameters, deserializationSchema, redisClient);
+    }
+
+    @Override
+    public void pollZsetToNext(List<String> keys, Collector<SeaTunnelRow> output)
+            throws IOException {
+        List<List<String>> zSetList = redisClient.batchGetZset(keys);
+        for (int i = 0; i < zSetList.size(); i++) {
+            for (String value : zSetList.get(i)) {
+                pollValueToNext(keys.get(i), value, output);
+            }
+        }
+    }
+
+    @Override
+    public void pollSetToNext(List<String> keys, Collector<SeaTunnelRow> output)
+            throws IOException {
+        List<Set<String>> setList = redisClient.batchGetSet(keys);
+        for (int i = 0; i < setList.size(); i++) {
+            for (String value : setList.get(i)) {
+                pollValueToNext(keys.get(i), value, output);
+            }
+        }
+    }
+
+    @Override
+    public void pollListToNext(List<String> keys, Collector<SeaTunnelRow> output)
+            throws IOException {
+        List<List<String>> valueList = redisClient.batchGetList(keys);
+        for (int i = 0; i < valueList.size(); i++) {
+            for (String value : valueList.get(i)) {
+                pollValueToNext(keys.get(i), value, output);
+            }
+        }
+    }
+
+    @Override
+    public void pollStringToNext(List<String> keys, Collector<SeaTunnelRow> output)
+            throws IOException {
+        List<String> values = redisClient.batchGetString(keys);
+        for (int i = 0; i < values.size(); i++) {
+            pollValueToNext(keys.get(i), values.get(i), output);
+        }
+    }
+
+    private void pollValueToNext(String key, String value, Collector<SeaTunnelRow> output)
+            throws IOException {
+        JsonNode node = JsonUtils.toJsonNode(value);
+        if (node.isTextual()) {
+            String text = node.textValue();
+            if (looksLikeJson(text)) {
+                try {
+                    node = JsonUtils.parseObject(text);
+                } catch (Exception e) {
+                    // do nothing
+                }
+            }
+        }
+
+        ObjectNode objectNode;
+        if (node instanceof ObjectNode) {
+            objectNode = (ObjectNode) node;
+        } else {
+            objectNode = JsonUtils.createObjectNode();
+            setValueInNode(objectNode, node);
+        }
+        objectNode.put("key", key);
```

Comment:
suggest this field use parameter, user can change this name. the default value is `key`